crates/matrix-sdk-base/src/store/mod.rs

```diff
@@ -380,6 +388,35 @@ pub trait StateStore: AsyncTraitDeps {
 #[cfg_attr(target_arch = "wasm32", async_trait(?Send))]
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 pub trait StateStoreExt: StateStore {
```

Comment:
I think we should also add a `remove_session_data` function to allow the clients to also remove the data from the StateStore.

Reply:
I was thinking of maybe taking an `Option` in `set_session_data`. The entry is then removed when it is `None`.

I believe this would simplify the use of this method because we usually have data that we want to set or remove as an `Option`.

Reply:
I would prefer the removal function to be a bit more explicit instead of having the set to be optional, I feel it makes the API more robust  (against missuses) and easier to read.
Also we already have functions in the API trait that set non optional values, and a dedicated remove function, and I think would be better to keep conforming to that standard.
(Check the media_content functions for example)

Reply:
I added the method. The issue with the new function signature is that `Option<T>` also implements `Serialize`, so one could try to insert a `None` value in the store. It would work, except in that case we would probably want to remove the entry from the store.

Reply:
My latest comment is not applicable anymore :slightly_smiling_face: